flixel/addons/editors/tiled/TiledTileSet.hx

```diff
@@ -92,23 +74,125 @@ class TiledTileSet
 				}
 				
-				var id:Int = Std.parseInt(node.att.id);
-				tileProps[id] = new TiledPropertySet();
-				tileProps[id].keys.set("id", Std.string(id));
-				for (prop in node.nodes.properties)
+				// read properties
+				properties = new TiledPropertySet();
+				for (prop in source.nodes.properties) {
+					properties.extend(prop);
+				}
+				
+				// read tiles properties
+				tileProps = new Array<TiledPropertySet>();
+				
+				for (node in source.nodes.tile)
+				{
+					if (!node.has.id)
+					{
+						continue;
+					}
+					
+					var id:Int = Std.parseInt(node.att.id);
+					tileProps[id] = new TiledPropertySet();
+					tileProps[id].keys.set("id", Std.string(id));
+					for (prop in node.nodes.properties)
+					{
+						tileProps[id].extend(prop);
+					}
+				}
+				
+				// read tiles images
+				tileImagesSources = new Array<TiledImageSet>();
+				
+				for (node in source.nodes.tile)
 				{
-					tileProps[id].extend(prop);
+					if (!node.has.id)
+					{
+						continue;
+					}
+					
+					var id:Int = Std.parseInt(node.att.id);
+					tileImagesSources[id] = new TiledImageSet();
+					tileImagesSources[id].keys.set("id", Std.string(id));
+					tileImagesSources[id].extend(node);
 				}
+				
+				if (tileWidth > 0 && tileHeight > 0)
+				{
+					numRows = 0;//cast(imgWidth / tileWidth);
+					numCols = 0;// cast(imgHeight / tileHeight);
+					numTiles = 0;// numRows * numCols;
+				}
+				
 			}
-			
-			if (tileWidth > 0 && tileHeight > 0)
+			else
 			{
-				numRows = cast(imgWidth / tileWidth);
-				numCols = cast(imgHeight / tileHeight);
-				numTiles = numRows * numCols;
+				//normal tileset
+				var node:Fast = source.node.image;
+				imageSource = node.att.source;
+				
+				var imgWidth = Std.parseInt(node.att.width);
+				var imgHeight = Std.parseInt(node.att.height);
+				
+				name = source.att.name;
+				
+				if (source.has.tilewidth) 
+				{
+					tileWidth = Std.parseInt(source.att.tilewidth);
+				}
+				if (source.has.tileheight) 
+				{
+					tileHeight = Std.parseInt(source.att.tileheight);
+				}
+				if (source.has.spacing) 
+				{
+					spacing = Std.parseInt(source.att.spacing);
+				}
+				if (source.has.margin) 
+				{
+					margin = Std.parseInt(source.att.margin);
+				}
+				
+				// read properties
+				properties = new TiledPropertySet();
+				for (prop in source.nodes.properties) {
+					properties.extend(prop);
+				}
+				
+				// read tiles properties
+				tileProps = new Array<TiledPropertySet>();
+				
+				for (node in source.nodes.tile)
+				{
+					if (!node.has.id)
+					{
+						continue;
+					}
+					
+					var id:Int = Std.parseInt(node.att.id);
+					tileProps[id] = new TiledPropertySet();
+					tileProps[id].keys.set("id", Std.string(id));
+					for (prop in node.nodes.properties)
+					{
+						tileProps[id].extend(prop);
+					}
+				}
+				
+				if (tileWidth > 0 && tileHeight > 0)
+				{
+					numRows = cast(imgWidth / tileWidth);
```

Comment:
Using `cast` for conversion to `Int` is unsafe / bad practice, use `Std.int()` instead.